File: src/types/git.ts
/** A single git commit. */
export interface Commit {
  hash: string;
  short_hash: string;
  message: string;
  author: string;
  timestamp: string;
  refs: string[];
}

/** Current working-tree / index status of the repository. */
export interface RepoStatus {
  branch: string;
  clean: boolean;
  modified: string[];
  staged: string[];
  untracked: string[];
}

/** Branch listing. */
export interface BranchList {
  branches: string[];
  current: string;
}

/** High-level information about the opened repository. */
export interface RepoInfo {
  path: string;
  name: string;
  branch: string;
  remote_url: string | null;
}

/** Result of a git merge operation. */
export interface MergeResult {
  success: boolean;
  conflicts: string[] | null;
}

/** A configured git remote. */
export interface Remote {
  name: string;
  url: string;
}

/** Result of a git pull operation. */
export interface PullResult {
  updated: boolean;
  new_commits: number;
  conflicts: string[] | null;
}

File: src-tauri/src/models/git.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Commit {
    pub hash: String,
    pub short_hash: String,
    pub message: String,
    pub author: String,
    pub timestamp: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct RepoStatus {
    pub branch: String,
    pub clean: bool,
    pub modified: Vec<String>,
    pub staged: Vec<String>,
    pub untracked: Vec<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct BranchList {
    pub branches: Vec<String>,
    pub current: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct RepoInfo {
    pub path: String,
    pub name: String,
    pub branch: String,
    pub remote_url: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct MergeResult {
    pub success: bool,
    pub conflicts: Option<Vec<String>>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Remote {
    pub name: String,
    pub url: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct PullResult {
    pub updated: bool,
    pub new_commits: u32,
    pub conflicts: Option<Vec<String>>,
}
